plugins/flatpak/gs-flatpak.c

```diff
@@ -3666,6 +3667,51 @@ app_has_local_source (GsApp *app)
 		(url != NULL && g_str_has_prefix (url, "file://"));
 }
 
+static gboolean
+gs_flatpak_allow_install_or_update (GsFlatpak *self,GsApp *app, GsAppList *list, gboolean is_update)
+{
+	g_autoptr (GFile) installation_dir = NULL;
+	g_autofree gchar *path = NULL;
+	GsApp *runtime, *update_runtime;
+	guint64 runtime_delta_size = 0;
+	guint64 min_free_space = 0;
+	guint i;
+	struct statvfs buf;
+
+	installation_dir = flatpak_installation_get_path (self->installation);
+	path = g_file_get_path (installation_dir);
+
+	if (statvfs (path, &buf) == -1) {
+		g_debug ("statvfs: Couldn't query disk space information");
+		return FALSE;
+	}
+
+	runtime = gs_app_get_runtime (app);
+	update_runtime = gs_app_get_update_runtime (app);
+	if (is_update && runtime != update_runtime) {
+
+		/* Correctly setup the runtime sizes if they are updatable */
+		guint64 runtime_size = gs_app_get_size_installed (runtime);
+		guint64 update_runtime_size = gs_app_get_size_installed (update_runtime);
+
+		if ((runtime_delta_size = update_runtime_size - runtime_size) > update_runtime_size)
+			runtime_delta_size = runtime_size - update_runtime_size;
+
+		gs_app_list_remove (list, update_runtime);
+	}
+
+	min_free_space = runtime_delta_size;
+	for (i = 0; i < gs_app_list_length (list); i++) {
+		GsApp *app_temp = gs_app_list_index (list, i);
+		min_free_space += gs_app_get_size_installed (app_temp);
```

Comment:
You should make sure that the `gs_app_get_size_installed` for every `app_temp` has a value (I don't know if at this point they already have it or need to be refined for it to be set up).